### PR TITLE
integration_test migration guide from flutter_driver

### PR DESCRIPTION
**Pull request by miquelbeltran**
_Description of what this PR is changing or adding, and why:_

In this PR I have created a migration guide for existing flutter_driver tests to integration_test, following the specs in https://github.com/flutter/website/issues/6171.

You can see a live version here: https://mb-flutter-dev-1.web.app/testing/integration-tests/migration

The guide presents the user with an example app with basic features. I decided to make it like a desktop app to change a bit from the typical mobile app format. The app uses a json file of bird names that is of public domain, and this file is not hosted but shipped with the example app. The source is located in this repo and can be accessed by any reader.

The app project contains three flutter_driver tests and as well the same tests migrated to use integration_test.

What I do is to show the original test in flutter_driver, explain how it works, and then do the same test using integration_test. Then repeat for each one of the three tests.

Some important things:

- I haven't modified the original integration-tests guide in flutter.dev/testing/integration-tests
- I also haven't added this guide to website menu.

_Issues fixed by this PR (if any):_

- https://github.com/flutter/website/issues/6171

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

cc. @RedBrogdon @johnpryan @domesticmouse

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

**Comment by kwalrath** [comment]
Quick drive-by comment... I expected the code samples to use [sentence-style comments](https://dart.dev/guides/language/effective-dart/documentation#do-format-comments-like-sentences) and output.

File: src/testing/integration-tests/migration.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+the `tester.pumpWidget` method.
+
+Then, you use `expect` with the matcher `findsOneWidget` to verify
+that the widget is displayed.
+
+<?code-excerpt "integration_test/main_test.dart (Test1)"?>
+```dart
+testWidgets('do not select any item, verify please select text is displayed',
+    (WidgetTester tester) async {
+  // load the PlantsApp widget
```
Comment (kwalrath) [inline src/testing/integration-tests/migration.md:108]: Here's an example of what I meant about using sentence-style comments:

```suggestion
  // Load the PlantsApp widget.
```

File: src/testing/integration-tests/migration.md [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+First you have to load the main app widget using
+the `tester.pumpWidget` method.
+
+Then, you use `expect` with the matcher `findsOneWidget` to verify
+that the widget is displayed.
+
+<?code-excerpt "integration_test/main_test.dart (Test1)"?>
+```dart
+testWidgets('do not select any item, verify please select text is displayed',
```
Comment (kwalrath) [inline src/testing/integration-tests/migration.md:106]: Here's an example of what I meant about using sentence-style output. (Ignore if this is against conventions, but by default we writers use Sentence-style for everything.)

```suggestion
testWidgets('Verify that the please select text is displayed. when no item is selected.',
```

File: src/testing/integration-tests/migration.md [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+
+As well, the tests verifies that the text "Please select a plant from
+the list." is not displayed anymore. 
+
+**flutter_driver**
+
+In `flutter_driver`, you use the `driver.tap` method to perform
+a tap over a widget using a finder.
+
+To verify that a widget is not displayed, you use the `waitForAbsert` method.
```
Comment (johnpryan) [inline src/testing/integration-tests/migration.md:139]:
```suggestion
To verify that a widget is not displayed, you use the `waitForAbsent` method.
```

**Comment by sfshaza2** [comment]
Hi, @miquelbeltran, nice to see you! This PR looks good, but is breaking on the null safety check. I just re-ran the job again, in case that fixes it. Can you take a look?

**Comment by miquelbeltran** [comment]
Thanks @sfshaza2! I updated the code excerpts and the checks are passing now

**Pull request merged**